explorer-client: Update listing for farm and reservation

grimpy opened the pull request:
Signed-off-by: Jo De Boeck <deboeck.jo@gmail.com>

grimpy committed:
explorer-client: Update listing for farm and reservation

Signed-off-by: Jo De Boeck <deboeck.jo@gmail.com>

zaibon reviewed (approved):
LGTM, but CI is not happy

muhamadazmy commented on tools/client/workloads.go:
```diff
@@ -19,8 +19,14 @@ func (w *httpWorkloads) Create(reservation workloads.Reservation) (id schema.ID,
-func (w *httpWorkloads) List(page *Pager) (reservation []workloads.Reservation, err error) {
+func (w *httpWorkloads) List(nextAction workloads.NextActionEnum, customerTid int64, page *Pager) (reservation []workloads.Reservation, err error) {
 	query := url.Values{}
+	if nextAction != 0 {
```
Actually nextAction of value 0 is a valid value. (check available values for the enum)
To work around this u can either do
- (check if value is < 0) and then caller has to pass of `-1` to donate (not set) (I don't like this very much)
- Or, you can change the input to `*NextActionEnum` then a nil value is possible, and then u only pass pointer to the value if u want to filter with next action. then u can do `if nextAction != nil` then `query.Set("next_action", *nextAction)`

muhamadazmy replied:
I just realized that the first option is not even possible because the Enum type is of size `uint8` so no support for minus value. Hence only possible option is using pointer value

grimpy committed:
explorer: Address issue with next_action filter

NextAction is an enum which should consider value 0
Hence in the client it is a pointer that can be passed nil

At the server side we nee to check if the query param is passed before
adding a filter (we can not check on != 0)

Signed-off-by: Jo De Boeck <deboeck.jo@gmail.com>

muhamadazmy commented on tools/explorer/pkg/workloads/types/reservation.go:
```diff
@@ -45,26 +46,23 @@ const (
-
 // Parse parese the query string
-func (rq *ReservationQuery) Parse(r *http.Request) mw.Response {
+func Parse(r *http.Request, filter ReservationFilter) (ReservationFilter, mw.Response) {
```
Why this return a mw.Response and not an error? I can see you only return bad request 
so It's better to return an error, and in the handler if u receive an error u wrap it in the bad request response.

grimpy replied:
no reaon at all corrected
